carcara/src/ast/mod.rs

```diff
@@ -739,6 +739,16 @@ impl Term {
         }
     }
 
+    /// Tries to extract a `bool` from a term. Returns `Some` if the term is an boolean
+    /// constant.
+    pub fn as_bool(&self) -> Option<bool> {
+        if let Term::Var(name, sort) = self {
+            Some(sort.as_sort() == Some(&Sort::Bool) && name == "true")
+        } else {
+            None
+        }
+    }
```

Comment:
This implementation is wrong. Any variable whose sort is not `Bool` will return `Some(false)`, implying that it is the boolean constant `false`. Instead, if the sort is not `Bool`, or the variable name is not `true` or `false`, you should return `None`:
```suggestion
    pub fn as_bool(&self) -> Option<bool> {
        match self {
            Term::Var(name, sort) if sort.as_sort() == Some(&Sort::Bool) => match name.as_str() {
                "true" => Some(true),
                "false" => Some(false),
                _ => None,
            },
            _ => None,
        }
    }
```